src/ecTyping.ml

```diff
@@ -1536,9 +1536,20 @@ let trans_if_match ~loc env ue (gindty, gind) (c, b1, b2) =
 
   List.map
     (fun (x, xargs) ->
-      if   sym_equal c x
-      then (cargs, Some b1)
-      else (List.map (fun ty -> (EcIdent.create "_", ty)) xargs), b2)
+      if sym_equal c x then 
+        (cargs, Some b1)
+      else 
+        (* Create a pattern C _ _ ... for each constructor *)
+        let wilds = List.map (fun _ -> EcLocation.mk_loc loc None) xargs in 
+        let pat = PPApp ((EcLocation.mk_loc loc ([], x), None), wilds) in
+        let b2 = 
```

Comment:
You can use `Option.to_list`

Reply:
This maps `Some b2 -> [b2]` but we need `Some b2 -> b2` so I don't think it applies here. But `Option.value` with `~default:[]` could do it.

Reply:
Ah yes. Whatever, we can merge then.